content/blog/2021/09/camel-kafka-consumer-changes/index.md

```diff
@@ -0,0 +1,68 @@
+
+## Motivation
+
+The ground work related to these changes can be traced back to the following set of issues reported on our Jira. Some of these issues were relatively new, but some have been reported for some time. In no particular order, we have:
+
+* [camel-kafka: possible corruption of idempotency messages when using KafkaIdempotentRepository](https://issues.apache.org/jira/browse/CAMEL-16914)
+* [camel-kafka: usage of deprecated method may cause Camel to block indefinitely](https://issues.apache.org/jira/browse/CAMEL-16928)
+* [camel-kafka: investigate alternatives to remove Thread.sleep from the record fetcher](https://issues.apache.org/jira/browse/CAMEL-16949)
+* [camel-kafka: commits may be skipped on stop or shutdown](https://issues.apache.org/jira/browse/CAMEL-16973)
+* [camel-kafka: improve the documentation about consumer streams vs consumers count](https://issues.apache.org/jira/browse/CAMEL-16981)
+* [camel-kafka -consumerCount vs consumerStreams problem with pool size](https://issues.apache.org/jira/browse/CAMEL-13215)
+* [camel-kafka: make the resume strategy configurable](https://issues.apache.org/jira/browse/CAMEL-16974)
+* [Seek to specific offset and KafkaConsumer access](https://issues.apache.org/jira/browse/CAMEL-13768)
+
+The main driver for these changes was our intention to move away from blocking consumer calls from the Kafka Consumer API, in order to leverage the modified APIs introduced on [KIP-266](https://cwiki.apache.org/confluence/display/KAFKA/KIP-266%3A+Fix+consumer+indefinite+blocking+behavior). Although the API changes by themselves are simple, the original behavior of the component consumer was not optimized for making the best use of them.
+
+## Concurrency Changes
+
+While reviewing and refactoring the code to implement the aforementioned set of changes, we used the opportunity to tweak the mechanism used for concurrently fetch records from Kafra. In previous versions, Camel had two configurations controlling the concurrency of the consumer threads: consumerStreams and consumersCount. The former was used to configure the size of the thread pool used by the consumers, whereas the latter configured the number of concurrent consumer threads used by the component. As part of these changes, the size of the thread pool is now defined by the number of concurrent consumers.
+
+## Resume Strategy
+
+This refactoring also allowed us to introduce one new feature that should provide greater flexibility for integrations willing to manage the resume process. It is now possible to define a custom resume strategy through the resumeStrategy configuration. A resume strategy can be recorded in the registry and referenced using the `resumeStrategy` configuration (i.e.: `resumeStrategy=#beanNameOnRegistry`).
```

Comment:
Can you configure the resume strategy on the component level, in case you want the same for all your Camel kafka endpoints?

Reply:
I'll have to try that one

Reply:
I was checking and it does not look like it supports setting it at the component level. 

Do you think I should make it clearer on the text (and, perhaps, log a record on Jira to investigate further for 3.13) ?

Reply:
Oh you can just create a JIRA for 3.13, the text is fine

Reply:
Thanks. I've logged CAMEL-16996 for that.